packages/browser/src/Session.ts

```diff
@@ -227,6 +227,27 @@ export class Session extends EventEmitter {
         this
       )
     );
+
+    // When a session is logged in, we want to track its ID in local storage to
+    // enable silent refresh.
+    this.on("login", () => {
+      // Store the current session ID specifically in 'localStorage' (i.e., not using
+      // any other storage mechanism), as we don't deem this information to be
+      // sensitive, and we want to ensure it survives a browser tab refresh.
+      window.localStorage.setItem(KEY_CURRENT_SESSION, this.info.sessionId);
```

Comment:
Is there a reason to not use the SessionInfoManager here? Or at least, to wrap the interaction with localstorage in a class?

Reply:
The reason for it being separate from the `SessionInfoManager` is because this is a very browser-specific mechanism, which doesn't make sense in a browser context: it is only useful for silent login on refresh. `SessionInfoManager`, on the other hand, is the abstraction layer on top of storage which is shared between the browser and node codebase.

It could be wrapped in a separate class, but the intent is that it is basically scoped to the `Session` class, so I'm not sure the extra indirection is required.

Reply:
Right, it's just that we're not able to mock it as a clear interface (we're mocking localstorage instead) and we're duplicating keys, plus there's a few other localStorage interactions that happen directly, and they're not all in one file.

Reply:
Yeah, some of these localStorage interaction are only required for a deprected workaround, and I'm currently investigating to see if anyone is still relying on it (I don't think so, and if not I'd want to remove it altogether). Otherwise, I think the only significant localStorage interaction is in `ClientAuthentication`, and I'll move it to the `Session` file so that it is all located in one place.

I don't know if there would be clear benefits in mocking it as a different interface, as this is explicitly relying on the browser local storage, as opposed to the `StorageUtility` we use otherwise, which is injected a storage instead.